start flanker task and local mock-up server

Floozutter opened the pull request:
```
The pvs-tasks directory contains a minimal Node.js server (Express) to
serve static files for the tasks. Usage help is in the README.md.
```

noah10 commented:
I have a few comments, mostly around organization (plus one around fragments):

- I'm not sure we need the pvs-tasks directory. I don't think anything is going to go into baseline that isn't a pvs-task.
- While the node/express stuff will definitely be helpful for local dev work, we should move it into a separate directory to make it easier to omit it from distribution. Maybe baseline/dev-tools/local-server ? Open to other ideas on the name(s), of course.
- If we move the node/express stuff, do we need a "public" directory? Shouldn't everything in baseline/client be public?

Regarding fragments, I see I should have spent time describing the big-picture architecture in our first conversation. All of these tasks are going to be served as a bundle from s3, not file-by-file from a "normal" web server, which means that you shouldn't need to call 'fetch' to get your html fragments - they will already be on the user's computer as part of the bundle. In fact, the entire fragment loader can go away, as this functionality is already provided by various bits of bundling infrastructure. You can just write something like:

`const intro_html = require('fragments/introduction.html')`

...and then pass intro_html as part of the relevant stimulus.

If you're not familiar with bundling (and related stuff like minifying, splitting, cache-busting, etc.), I'd recommend taking a look at https://www.simplethread.com/javascript-modules-and-code-bundling-explained/ . I'd like to use webpack as our bundler; a reasonable intro to it is https://github.com/petehunt/webpack-howto . Webpack provides [html-loader](https://webpack.js.org/loaders/html-loader/#root) , which essentially performs the same role as your fragment loader. The documentation is a little dense, though, so https://robkendal.co.uk/blog/use-webpack-and-require-context-to-load-html-files might provide a clearer example. (Although in this case I think you can ignore his require-context stuff - for three files it's not really necessary.)

All this being said, after talking with Mara about the site registration process I'm beginning to think it might be complicated enough to justify using [Vue.js](https://vuejs.org/). Take a look at that (keeping in mind edge case scenarios like needing to put someone through a login flow before you can upload their data from an experiment) and let me know whether you'd rather use that or just go with webpack for the experiments you'll be working on. I'm on the fence as to whether to use Vue for the other site elements (reg, auth, forgot password, etc.) or not.

Regarding the code itself: I like it! I found it very easy to read and I like that you're careful to use const where you can. The only thing I was a little unclear on was what purpose the public/index.html file served. Is this ultimately going to be a list of experiments that users pick from or something?

Floozutter commented:
Your suggestions for structuring the project are way better than whatever I started doing! I got careless because I just wanted something running. I moved all the local server stuff into `baseline/dev-tools/local-server/`, so now `baseline/client/` should hold the static files directly. I also moved the `public/index.html` file there. I made it just as a convenience so I could navigate in the browser to each task from a single page.

I'm super relieved to hear that I don't have to write any code to load the HTML fragments! My current implementation felt flimsy. I'll read the links you sent me about bundling and get it working on the local server so I can tear out `fragload.js`.

I'll check out Vue.js as well. What alternatives are you considering?

About the code, I'm starting to think that I should pick up TypeScript ASAP. I'm starting to feel a little paranoid without a type checker.

noah10 commented:
Right now I'm thinking either Vue or roll our own (i.e., like what you've done so far) and just bundle it with webpack. I used Angular for the HRV project and felt that it brought extra complexity (compared to Vue) without corresponding benefit, so I don't want to use that again. If you've used something you like I'm willing to take a look, but we should definitely pick something at least one of us has used. I've used Vue for several projects - you can take a look at https://github.com/EmotionCognitionLab/flywheel/tree/master/utils/mark-inputs . Unfortunately you need Flywheel credentials to actually use the app, so you won't be able to see it in action, and it may not make much sense without that.

And I'm fine with typescript. It means a bit of extra complexity to get the transpiler working as part of the bundling process, but go for it if you want to do that. (I would like to be consistent, though, so everything in baseline/client should either be stock JS or typescript.)

Floozutter commented:
I couldn't find any type declaration files for jsPsych (in their package or in Definitely Typed), so I'm realizing now that using TypeScript may not be worth that extra complexity. I'll continue using vanilla JS for now.

noah10 commented on baseline/client/flanker/frag/instruction.html:
```diff
@@ -0,0 +1,5 @@
+<h1>Instruction:</h1>
+For each trial in this task, 5 arrows will be displayed on the screen. <br>
+Your goal is to press the button that corresponds to the direction of the <i>center arrow</i>. <br>
```
Very, very minor nit: These days <em> is generally preferred to <i> for this sort of usage. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/em

noah10 commented on baseline/client/flanker/index.html:
```diff
@@ -0,0 +1,14 @@
+    <script defer src="/js/jspsych-html-keyboard-response.js"></script>
+    <script defer src="/js/fragload.js"></script>
+    <script defer src="/flanker/task.js"></script>
+    <link rel="stylesheet" href="/css/jspsych.css">
```
I don't see a jspsych.css file checked in anywhere.

noah10 reviewed (approved):
Other than the replacement of the fragment loader with webpack's html-loader (and probably the replacement of the local express server with webpack-dev-server) that we've already discussed, this looks good.
